frontend/smart-kickers-game/src/App.js

```diff
@@ -24,6 +30,45 @@ function App() {
     };
   }, []);
 
+  const ScorePrevious = (value) => {
+    const ref = useRef();
+    useEffect(() => {
+      ref.current = value;
+    });
+    return ref.current;
+  };
+
+  const prevBlueScore = ScorePrevious(blueScore);
+  const prevWhiteScore = ScorePrevious(whiteScore);
+
+  useEffect(() => {
+    if (prevBlueScore > blueScore) {
+      const indexOfLastBlue = goalsArray.indexOf(goalsArray.findLast((e) => e.teamID === TeamID.Team_blue));
+      goalsArray.splice(indexOfLastBlue, 1);
+    } else {
+      goalsArray.push(new Goal(TeamID.Team_blue, 'time: ' + minutes + ':' + seconds));
```

Comment:
Wouldn't it be nice to format the time somehow? You could use `padStart` method to add the leading zeroes to a value with single digit.

Reply:
Done